### PR TITLE
WIP - Performance enhancement changes

### PR DESCRIPTION
**Pull request by shtripat**
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
Snippet of utilization for node-agent post these changes (output from below command)

```
while true;
  do ps -p $(echo $(ps aux | grep [t]endrl-node-agent | awk '{print $2}') | sed 's/ /,/') -o %cpu,%mem,cmd -h;
  sleep 5;
done
```

```
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
 5.6  3.5 /usr/bin/python /usr/bin/tendrl-node-agent
```

File: tendrl/commons/flows/create_cluster/ceph_help.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ def install_packages(parameters):
     mon_ips = []
     osd_ips = []
     for node, config in parameters["Cluster.node_configuration"].iteritems():
-        if "mon" in config["role"].lower():
```
Comment (r0h4n) [inline tendrl/commons/flows/create_cluster/ceph_help.py]: these changes not required, create and import flows arent called often, And what if someone passes "mOn" or some value which we havent checked for

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 2]
```diff
@@ -215,7 +215,7 @@ def run(self):
                             )
                         )
 
-            if "ceph" in sds_name.lower():
```
Comment (r0h4n) [inline tendrl/commons/flows/import_cluster/__init__.py]: not required

File: tendrl/commons/objects/__init__.py [hunk 3]
```diff
@@ -75,94 +75,27 @@ def load_definition(self):
...
-                                )
-                            )
-            try:
-                NS._int.wclient.write(item['key'], item['value'], quorum=True)
-            except (etcd.EtcdConnectionFailed, etcd.EtcdException):
-                NS._int.wreconnect()
-                NS._int.wclient.write(item['key'], item['value'], quorum=True)
-        if ttl:
-            etcd_utils.refresh(self.value, ttl)
+        key = self.value + "/data"
```
Comment (r0h4n) [inline tendrl/commons/objects/__init__.py]: whats happening here? Seems like a major change to the way tendrl objects are stored in etcd

Comment (shtripat) [inline tendrl/commons/objects/__init__.py]: Yes. We write the json representation of the object as single field here. While read the single field is weaved back into the object instance.

**Comment by r0h4n** [comment]
As per https://www.redhat.com/archives/tendrl-devel/2017-July/msg00004.html

Please send this PR to a "spec/$spec_name" branch or against "master" if it's a bugfix

**Comment by shtripat** [comment]
Do we have a spec for this already? I dont think so.

**Comment by r0h4n** [comment]
@shtripat you want to volunteer writing the spec based on our past discussions?

**Comment by shtripat** [comment]
Can start one tomorrow may be :)

**Pull request closed without merging**